pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -623,6 +643,20 @@ func (p *PriorityQueue) MoveAllToActiveOrBackoffQueue(event framework.ClusterEve
 	p.movePodsToActiveOrBackoffQueue(unschedulablePods, event)
 }
 
+// moveAllToActiveOrBackoffQueue moves all pods from unschedulablePods to activeQ or backoffQ.
+// This function adds all pods and then signals the condition variable to ensure that
+// if Pop() is waiting for an item, it receives the signal after all the pods are in the
+// queue and the head is the highest priority pod.
```

Comment:
Add a comment saying:

```
// NOTE: this function assumes a lock has been acquired in the caller.
```

Reply:
And in the uppercase `MoveAllToActiveOrBackoffQueue()` function above (L634), call this function instead of duplicating the logic.